NuGet/ServerCommon
# Move from WindowsAzure.ServiceBus to Azure.Messaging.ServiceBus

Address https://github.com/NuGet/Engineering/issues/5011

Summary of changes:
1. Remove WindowsAzure.ServiceBus dependency, add Azure.Messaging.ServiceBus dependency
2. Introduce `IReceivedBrokeredMessage`. This mirrors a similar split in the send and receive message models in the new SDK.
   - This is a better data model IMO since some things are immutable on received messages and the models can express that.
3. Add several shim-behaviors, to ensure full back-compat
   - Generate a random message ID on the client side (new SDK doesn't so this, partitioned topics require this)
   - Add support for `sb://namespace/` style connection strings (old SDK supports it, new SDK does not)
   - Serialize message content using XML data contract. I dislike this but we must do this to handle messages sent by old SDK

Testing strategy:
1. Full deployment to DEV
    - Gallery, internal jobs, OSS jobs
    - Test send-receive compat (sender on new SDK, receiver on old SDK and vice versa)
    - Push steady stream of validations and reboot VMs (non-graceful message handling)
2. Full deployment in INT
3. PROD strategy:
    - Deploy single region of SB listeners and observe telemetry
    - Wait 48 hours
    - Deploy 2nd region of SB listeners
    - Wait 24 hours
    - Deploy gallery
